### ISSUE TITLE
AI - Image Crop Recognition - Model Development

### ISSUE THREAD

**Issue by chebibj** [bahrain-bp/mun-aquacrop#66]
2. Model Development
2.1 Training the AI Model
- Pretrained Model: Use MobileNet (or another pre-trained model available in SageMaker
JumpStart) as a base model for transfer learning.
- Why MobileNet? MobileNet is the default. It is lightweight, efficient, and well-suited for
mobile and real-time inference, which aligns with the application’s requirements.
- Fine-Tuning: Fine-tune the MobileNet model with your labeled dataset of crop images.
This step adapts the model to your specific crop types and stages.
Steps:
1. Data Loading:
- Load the image data from **S3** into **SageMaker Studio**. Use a data input pipeline
that handles batching, shuffling, and data augmentation (e.g., random rotations or flips) to
make the model more robust.
2. Model Architecture:
- Use a pre-trained MobileNet as the backbone for the convolutional layers.
- Add a few fully connected layers at the end for classification, corresponding to the 30
possible outputs (10 crop types × 3 stages).
3. Splitting the Data:
- Training Set: Use 70% of the data for training.
- Validation Set: Use 20% of the data for validating the model during training to monitor
performance
- Test Set: Use 10% of the data for final evaluation after the model is trained.
4. Fine-Tuning:
- Unfreeze the later layers of the MobileNet model to allow the model to adapt to your
dataset.
- Use a learning rate scheduler to gradually decrease the learning rate during training for
better convergence.
5. Training Configuration:
- Train the model using the AWS SageMaker built-in image classification algorithm or by
implementing custom code with TensorFlow or PyTorch.
- Use distributed training if the dataset is large to speed up training time.
2.2 Model Evaluation
Evaluation Steps:
1. Validation Set: After training, evaluate the model using the validation set.
2. Testing: Finally, test the model with the test set to simulate real-world performance.
2.3 Model Deployment
- Deploy the Fine-Tuned Model:
- Once the model is trained, deploy it to a SageMaker real-time endpoint. This allows the
mobile app to interact with the model via API calls.
- Model Endpoint:
- Expose a **REST API** through **SageMaker hosting** where the mobile app can send
requests with images.
- The response will include the predicted ‘crop type’, ‘growth stage’, and ‘confidence
score’.
Sample Output:
```json
{
"crop_type": "Tomato",
"growth_stage": "Mid",
"confidence": 0.95
}
```
